packages/loader/container-utils/src/error.ts

```diff
@@ -120,9 +89,38 @@ export const extractSafePropertiesFromMessage = (message: ISequencedDocumentMess
     });
 
 /**
- * Conditionally coerce the throwable input into a DataProcessingError.
+ * Normalize the throwable input into a Fluid Error annotated as a DataProcessingError.
+ * @param originalError - Potentially untrusted error to be converted/annotated.
+ * @param errorCodeIfNone - errorCode identifying the call site, to be used if the originalError has no error code.
+ * @param message - Sequenced message (op) to include info about via telemetry props
+ * @returns An error annotated as dataProcessingError in a variety of ways to support telemetry analysis
  */
-export const CreateProcessingError = DataProcessingError.wrapIfUnrecognized;
+export function CreateProcessingError(
+    originalError: unknown,
+    errorCodeIfNone: string,
+    message: ISequencedDocumentMessage | undefined,
+): IFluidErrorBase {
+    // Temp back-compat for Telemetry systems expecting the errorType to be dataProcessingError
+    const overrideNormalizedErrorCode = !isValidLegacyError(originalError);
+
+    const error = normalizeError(originalError, {
+        errorCodeIfNone,
+        props: { dataProcessingError: 1 },
+    });
+
+    // canRetry is a property that various code checks for.  Should be false in this case.
+    Object.assign(error, { canRetry: false });
+
+    if (overrideNormalizedErrorCode) {
+        Object.assign(error, { errorType: ContainerErrorType.dataProcessingError });
+    }
+
+    if (message !== undefined) {
+        error.addTelemetryProperties(extractSafePropertiesFromMessage(message));
```

Comment:
will the above also make it to the telemetry properties

Reply:
specifically errorType and canRetry

Reply:
also why add these here rather than via normalize error?

Reply:
ah crap, I screwed this up, good catch.  Hmm.